### PR TITLE
Add Tests and more Table Borders operations

### PR DESCRIPTION
**Pull request by Andres-CT98**
### Description:

Improve the ability to modify table borders with more operations.

### Fix

* Add more Table border functions to modify and apply changes to table borders to selected cells.

Simplified the operations by having only one function to apply borders to cells. It receives an array of the borders to set, the border format, and the cell. In the case of no border, it is set to '', the attribute is eventually deleted. The functionality to modify the perimeter of the selection has also been implemented as its own function.

* Modified Demo site to Include Table Borders functionality: Left, Right, Top, Bottom, Inside, Outside, and No borders operations added.

* Added tests for Table Border related changes and new functionality.

### Warnings and implications

- None

### CODE REVIEW

File: packages-content-model/roosterjs-content-model-editor/lib/publicApi/table/applyTableBorderFormat.ts [hunk 1]
```diff
@@ -64,56 +83,182 @@ export default function applyTableBorderFormat(
...
 
             if (sel) {
-                switch (operation) {
-                    case 'AllBorders':
-                        for (let rowIndex = sel.firstRow; rowIndex <= sel.lastRow; rowIndex++) {
-                            for (let colIndex = sel.firstCol; colIndex <= sel.lastCol; colIndex++) {
-                                const cell = tableModel.rows[rowIndex].cells[colIndex];
-                                // Format cells - All borders
-                                applyAllBorderFormat(cell, borderFormat);
+                const operations = [operation];
```
Comment (JiuqingSong) [inline packages-content-model/roosterjs-content-model-editor/lib/publicApi/table/applyTableBorderFormat.ts:86]: Suggest to add a type declaration:

const operatons: BorderOperations[] = [operation];

File: packages-content-model/roosterjs-content-model-editor/lib/publicApi/table/applyTableBorderFormat.ts [hunk 2]
```diff
@@ -64,56 +83,182 @@ export default function applyTableBorderFormat(
...
+                            // Inner cells
+                            sel.firstCol++;
+                            sel.firstRow++;
+                            sel.lastCol--;
+                            sel.lastRow--;
+                            operations.push('AllBorders');
+                            break;
+                        case 'OutsideBorders':
+                            // Format cells - Outside borders
+                            operations.push('TopBorders');
```
Comment (JiuqingSong) [inline packages-content-model/roosterjs-content-model-editor/lib/publicApi/table/applyTableBorderFormat.ts:250]: I like this design.

File: packages-content-model/roosterjs-content-model-editor/lib/publicTypes/enum/BorderOperations.ts [hunk 3]
```diff
@@ -5,4 +5,32 @@ export type BorderOperations =
     /**
      * Apply border format to all borders
      */
-    'AllBorders';
+    | 'AllBorders'
```
Comment (JiuqingSong) [inline packages-content-model/roosterjs-content-model-editor/lib/publicTypes/enum/BorderOperations.ts:8]: Not a big problem. All existing string literal values are started with lower case. There is no strong reason but I still want to make them consistent, either all lower case or all upper case.

Comment (Andres-CT98) [reply packages-content-model/roosterjs-content-model-editor/lib/publicTypes/enum/BorderOperations.ts:8]: You mean for example changing AllBorders to allBorders?

Comment (JiuqingSong) [reply packages-content-model/roosterjs-content-model-editor/lib/publicTypes/enum/BorderOperations.ts:8]: Yes.

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**